.github/workflows/go.yml

```diff
@@ -10,8 +10,25 @@ permissions:
   contents: read
 
 jobs:
+  conditional-changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      doc: ${{ steps.filter.outputs.doc }}
+    steps:
+    - uses: actions/checkout@v3
+    - uses: dorny/paths-filter@v2
+      id: filter
+      with:
+        filters: |
+          doc:
+            - 'site/**'

```

Comment:
> I have confirmed that we need to add the condition for this job as well
> 
> ```yaml
> //  .github/workflows/apidiff.yml
> 
> jobs:
>   go-apidiff:
>     name: Verify API differences
>     runs-on: ubuntu-latest
> ...
> ```

Yes, I was looking through that and intentionally skipped that. As its single job in a different workflow so adding this condition would mean adding the same conditional job there too. And that will always run to check condition and then make decision to skip that single api diff job or not. I took judgement call to skip doing this there as incase of code changes we will be spending additional time and resources to the same workflow and incase of document changes a single job will run anyways crawling though the code. Im open to your suggestions and we surely can integrate it there as well.

Reply:
> Thanks @sarab97 for the PR!
> 
> I suppose, we have to ignore:
> 
>     * somePath/*_test.go files
> 
>     * somePath/*.md files
> 
>     * and somePath/.gitignore files
> 
> 
> I'm not sure what "Diff the API" means referred on the issue page, will ask further clarification on this and get back to you

Im not sure adding `*_test.go` files falls under doc changes scope. However `*md` file does make sense.  I was thinking maybe we first stablize this then we can easily add any more conditions as per need. 
"diif the api" refers to that apidiff workflow

Reply:
> Let's add the condition there in case apidiff workflow expands in the future. Or maybe we can make condition check job generic and can be referenced from these workflows, maybe we can use these strategies:

The apidiff is a single action workflow having a job. A workflow has single purpose. So checking for api diff that wont be expanding. 

I have looked though link you provided. Those do seem interesting approach but have their own drawbacks like rerunning it again for each call which will be inefficient in our case.  
I would prefer keeping this simple rather than overcomplicating.

> Ah you are right, I misunderstood the requirements, you don't need to ignore *_test after all. Speaking of detail of the check, does a code comment change counts as code change or docs change?

We dont need to go in so deep wondering about comment change. We wont be having any way to check that. And even if possible it would be too much of effort for too little of a reward.

Reply:
Yeah @antoooks I left it out to be done in future. But sure I will add both conditions.